airflow/jobs/scheduler_job.py

```diff
@@ -779,6 +779,22 @@ def _execute(self) -> None:
                 self.log.exception("Exception when executing DagFileProcessorAgent.end")
             self.log.info("Exited execute loop")
 
+    def _update_dag_run_state_for_paused_dags(self):
```

Comment:
since no callbacks is executed in the LocalTaskJob, we need to rely on the scheduler to check and update the paused dag runs.

Reply:
I'm not too happy about this change -- the point of having this in LocalTaskJob is so that the state of a dag run updates quickly after the task finishes.

By moving it to the scheduler we've lost that.

Instead of moving this to the scheduler, could we move it "down" in to the raw task process instead? That or I wonder if there is some way with AIP-44 to add a callback from the LTJ if needed.

Final option: _If_ there are no callbacks we can still safely do this here, so maybe we only need to do this in the scheduler in case of callbacks?

Reply:
good point on having the LocalTaskJob to mark the dag run quickly. my concern is that it leaks the responsibility of the scheduler, also there is chance that the LTJ fails before marking the dag run state, which leaves the unhandled dag run states (this could be rare). also, since the dag is paused, it would be a good deal if the dag run state is updated a little bit late.

as we discussed in this email thread: `[DISCUSSION] let scheduler heal tasks stuck in queued state`, we will need to define the responsibility of each component in airflow (scheduler, executor, LTJ (airflow run --local), `airflow run --raw`) in terms of the state machine. 

we can have a thorough discussion there. let me know your thoughts.

Reply:
I quite agree with no need to update pasused tasks state immediately - this makes this part of the changfe far more appealing. Since this change is merged - @ashb - if you stil have some concerns, we can continue discussing here and maybe make a follow-up change in case we think of a scenario where it might be problematic.